features/eolearn/features/doubly_logistic_approximation.py

```diff
@@ -0,0 +1,84 @@
+"""
+Module for calculating doubly logistic approximation.
+"""
+import itertools as it
+import numpy as np
+from scipy.optimize import curve_fit
+from eolearn.core import EOTask
+
+
+def doubly_logistic(middle, initial_value, scale, a1, a2, a3, a4, a5):
+    # pylint: disable=invalid-name,locally-disabled
+    """
+    Function that is passed to scipy.optimize
+    """
+    return initial_value + scale * np.piecewise(middle, [middle < a1, middle >= a1],
+                                                [lambda y: np.exp(-((a1 - y) / a4) ** a5),
+                                                 lambda y: np.exp(-((y - a1) / a2) ** a3)])
+
+
+class DoublyLogisticApproximationTask(EOTask):
+    """
+    EOTask class for calculation of doubly logistic approximation on each pixel for a feature. The task creates new
+    feature with the function parameters for each pixel as vectors.
+    :param feature: A feature on which the function will be approximated
+    :type feature: str
+    :param new_feature: Name of the new feature where parameters of the function are saved
+    :type new_feature: str
+    :param initial_parameters: Initial parameter guess
+    :type initial_parameters: List of floats length 7 corresponding to each parameter
+    :param mask_feature: A feature used as a mask for valid regions. If left as None the whole patch is used
```

Comment:
Perhaps it's clearer if this parameter is `valid_mask` or something, we've had issues in the past due to the confusion.